### PR TITLE
include/compat.h: make pthread_get_name_np work when available

### PR DESCRIPTION
**Pull request by wjwithagen**
FreeBSD will have a pthread_get_name_np() in libpthread
Code for that was submitted

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/include/compat.h [hunk 1]
```diff
@@ -159,6 +159,11 @@
 
 #if defined(HAVE_PTHREAD_GETNAME_NP)
   #define ceph_pthread_getname pthread_getname_np
+elif defined(HAVE_PTHREAD_GET_NAME_NP)
```
Comment (tchaikov) [inline src/include/compat.h]: this does not compile.

File: src/include/compat.h [hunk 2]
```diff
@@ -159,6 +159,11 @@
...
+  /* Fix a small name diff and return 0 */
```
Comment (tchaikov) [inline src/include/compat.h]: please drop this comment. it's literally a repeating of following two lines.

**Comment by wjwithagen** [comment]
@tchaikov 
Running with a recent kernel:
```
-- Looking for pthread_get_name_np
-- Looking for pthread_get_name_np - found
```
http://cephdev.digiware.nl:8180/jenkins/job/ceph-master/2469/consoleFull

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-08-20_02:36:43-rados-wip-kefu-testing-2018-08-19-2329-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-21_09:24:49-rados-wip-kefu-testing-2018-08-19-2329-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-21_13:03:23-rados-wip-kefu-testing-2018-08-19-2329-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-21_15:56:38-rados-wip-kefu-testing-2018-08-21-1839-distro-basic-mira/

**Pull request merged**